### PR TITLE
Migrate `digamma` and `digamma_` from the TH to Aten (CPU)

### PR DESCRIPTION
**Pull request by ifedan**
https://github.com/pytorch/pytorch/issues/24690

### CODE REVIEW

**Comment by ifedan** [comment]
image

export OMP_NUM_THREADS=1
export MKL_NUM_THREADS=1

**Comment by ifedan** [comment]
**Singlethread test**
image

**Comment by ifedan** [comment]
image
export OMP_NUM_THREADS=1
export MKL_NUM_THREADS=1

File: aten/src/TH/generic/THTensorMoreMath.cpp [hunk 1]
```diff
@@ -1161,7 +1160,14 @@ LAB_IMPLEMENT_VECTORIZED_FUNCTION(sigmoid,TH_MATH_NAME(TH_sigmoid),HYPER_TH_OMP_
 
 void THTensor_(polygamma)(THTensor *r_, int64_t n, THTensor *t) {
   switch (n) {
-    case 0: THTensor_(digamma)(r_, t); break;
+    case 0: {
+      THTensor_(resizeAs)(r_, t);
+      at::Tensor r__wrap = THTensor_wrap(r_);
+      at::Tensor t__wrap = THTensor_wrap(t);
+      at::Tensor result = at::native::digamma(t__wrap);
```
Comment (VitalyFedyunin) [inline aten/src/TH/generic/THTensorMoreMath.cpp]: You can use `_out` function and avoid additional copy (bet benchmarks showing slow down because of it)

**Comment by ifedan** [comment]
Will close this PR and one a new one with digamma/trigamma/polygamma

**Comment by ifedan** [comment]
https://github.com/pytorch/pytorch/pull/25048

**Pull request closed without merging**